src/services/mapService.js

```diff
@@ -0,0 +1,63 @@
+export const mapService = {
+  mapSettings: {
+    defaultWidth: 360,
+    defaultHeight: 180,
+    scale: 1.5
+  },
+
+  getMapElement(){
+    return document.getElementById("map");
+  },
+
+  getMapWidth() {
+    return this.mapSettings.defaultWidth * this.mapSettings.scale;
+  },
+
+  getMapHeight(){
+    return this.mapSettings.defaultHeight * this.mapSettings.scale;
+  },
+
+  translateLongitudeToCanvasPx(longitude){
+    let x = (longitude + this.mapSettings.defaultWidth / 2) * this.mapSettings.scale;
+    return x;
+  },
+
+  translateLatitudeToCanvasPx(latitude){
+    let y = ((this.mapSettings.defaultHeight / 2) - latitude) * this.mapSettings.scale;
+    return y;
+  },
+
+  initMap(){
+    const mapCanvas = this.getMapElement();
+    const ctx = mapCanvas.getContext("2d");
+    ctx.canvas.width = this.getMapWidth();
+    ctx.canvas.height = this.getMapHeight();
+
+    // draw some coffee shops
+    this.drawCoffeeShopAt(ctx, this.translateLongitudeToCanvasPx(25), this.translateLatitudeToCanvasPx(45));
```

Comment:
Magic numbers can cause trouble when debugging or to just understand the code, as the reader needs to think what this value might represent. Can you improve it?

Reply:
Yes, eventually this `drawCoffeeShopAt` method will receive the longitude and latitude from the API, I can send the function params as they come from API, and then I can translate them so that they can be drawn inside the function.